build.gradle

```diff
@@ -349,9 +348,10 @@ appengine {
                 "-Dblobstore.backing_store=appengine-generated"]
     }
     deploy {
-        Node appengineWebXml = new XmlParser().parse('src/main/webapp/WEB-INF/appengine-web.xml')
-        project = appengineWebXml.application.text()
-        version = appengineWebXml.version.text()
+        String appengineWebXmlPath = "${projectDir}/src/main/webapp/WEB-INF/appengine-web.xml"
+        Node appengineWebXml = new File(appengineWebXmlPath).exists() ? new XmlParser().parse(appengineWebXmlPath) : null
+        project = appengineWebXml == null ? "teammates-john" : appengineWebXml.application.text()
+        version = appengineWebXml == null ? "4-18" : appengineWebXml.version.text()
```

Comment:
Can explain the purpose of these changes?

Reply:
Allow me to bring you to this line which is at the end of the Eclipse setup instructions:

> **Note**: It is not encouraged to run Gradle tasks via Eclipse.

1. When running via Buildship, Gradle's working directory is Eclipse's (e.g. in Mac `.../Eclipse.app/Content/MacOS`) instead of the project's root directory. This causes some tasks not to be runnable, particularly those dependent on relative path. Above is one of them, which is why I needed to append `${projectDir}`. If you're interested, `checkstyle` is another due to the presence of the suppression filter.
1. Because of that, the above block will be evaluated and when `appengine-web.xml` does not exist, the line `new XmlParser().parse(...)` will return an exception. This is enough to prevent Eclipse from importing the project (if not fixed), and to stop Travis build (if fixed just as is). To make it a full fix, the value `appengine-web.xml` needs to be configured based on the presence of the `appengine-web.xml`.
1. Why does this happen only now? Because `createConfigs` is now wrapped in `doLast`. Without it, `createConfigs` is _always_ (wrongly) run and this has unknowingly saved us from the problem of non-existent `appengine-web.xml` that should have been introduced. In Eclipse, due to reason (1) up there, every single "Refresh Gradle project" will cause the config files to be reset.

Reply:
Can we proceed to set `project` and `version` only if the file exists? We don't expect anyone to run `appengineDeploy` when the file `appengine-web.xml` doesn't exist, right?

Reply:
That's what I want but that's not how Gradle works...

Reply:
Set to `null`?